roachtest: add roachperf stats

andrewbaptist opened the pull request:
This commit adds the creation of a stats.json file after the run. This file is collected by the testing framework to make graphs on the roachperf dashboard.

Fixes: #128393

Epic: none

Release note: None

Commit:
roachtest: query throughput using SQL for perturbation tests

Previously the rate for the cluster was collected from the workload
nodes. In an effort to switch away from the dependence on the roachtest
node in the critical path, this commit changes all the perturbation/*
tests to use the rate from the cluster nodes directly.

Epic: #129671

Release note: None

Commit:
workload: add parsing for JSON

Previously there was code to output JSON but no parsing utility. This
commit adds parsing.

Epic: #129671

Release note: None

Commit:
roachtest: use the rate from the log files for perturbation/* tests

Previously we would track the throughput and latency from the listener
on the nodes directly. This had problems in certain cases and added a
new failure mode. This change switches to pulling the rate from the log
files after the test completes.

Epic: #129671

Release note: None

Commit:
roachtest: correctly wait for errors in perturbation tests

Previously the perturbation/* tests did not check for errors in the
workers it launched. This made it hard to track down bugs or problems
with running the workload.

Epic: #129671

Release note: None

Commit:
roachtest: make Cluster an embedded struct in perturbation/* tests

Having the struct embedded makes it easier to use.

Epic: none

Release note: None

Commit:
roachtest: add roachperf stats for perturbation/* tests

This commit adds the creation of a stats.json file after the run. This
file is collected by the testing framework to make graphs on the
roachperf dashboard.

Fixes: #128393

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130262)
<!-- Reviewable:end -->

kvoli reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with the commit message updated.

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130262#-:-O6MEnzsAx4QYRtMv-9G:b-cycx6t)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @andrewbaptist, @renatolabs, and @srosenberg)

___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130262#-O6MEaUV9VwpX7mIVpjB):*
For which test?


<!-- Sent from Reviewable.io -->

Commit:
roachtest: auto-detect latency for profile collection

Previously the latency for profileTopStatements needed to be manually
passed in. This commit changes it to detect automatically based on 10x
the P99 latency. This is only used in the perturbation/* tests today.

Epic: #129671

Release note: None

andrewbaptist commented:
TFTR! Crossing fingers that all the graphs are populated on Monday!

bors r=kvoli
